web/src/components/vis/ScreePlotTile.vue

```diff
@@ -1,42 +1,42 @@
-<script>
+<script lang="ts">
+import { defineComponent, computed, ref } from '@vue/composition-api';
 import ScreePlot from '@/components/vis/ScreePlot.vue';
 import VisTile from '@/components/vis/VisTile.vue';
-import plotData from './mixins/plotData';
-
-export default {
-  components: {
-    ScreePlot,
-    VisTile,
-  },
-
-  mixins: [plotData('pca')],
+import usePlotData from './use/usePlotData';
 
+export default defineComponent({
   props: {
     id: {
-      required: true,
       type: String,
+      required: true,
     },
   },
 
-  data() {
+  components: { ScreePlot, VisTile },
+
+  setup(props) {
+    const id = computed(() => props.id);
```

Comment:
Is the point of this to protect the `usePlotData` function by wrapping the prop in a readonly ref?

You'll probably want to mark this as a TODO, since in Vue 3 it will be fine to pass `props.id` directly to `usePlotData`

Reply:
I was basically trying to make `props.id` reactive. `props` is reactive, but `props.id` is just a string, so if you navigate to a different dataset `props.id` will update, but that won't propagate from here to `usePlotData`.
If there's a better way to make it reactive that would be awesome, this feels gross.